0xProject/0x-monorepo
# asset-swapper: RFQ-T firm quotes

## Description

These changes support https://github.com/0xProject/0x-api/pull/162

Every commit is crafted to be completely independent and atomic. I suggest you review them one at a time, in order to give better context, rather than just reviewing all of the changes at once.

## Testing instructions

While there aren't any tests here yet, this code is currently being exercised by a 0x-api test, which you can see in the RFQ-T commit at https://github.com/0xProject/0x-api/pull/162 , and which is using the `asset-swapper`, and `contract-addresses` packages from this branch.

The deployment of ERC20BridgeSampler was necessary not just for the RFQ-T test but also for the vanilla `/swap/v0/quote` endpoint test as well.  Before that deployment, the `market_operation_utils` code was reverting when calling into the sampler, resulting in a HTTP 400 response from the 0x API.

## Types of changes



* Bug fix (non-breaking change which fixes an issue)

* New feature (non-breaking change which adds functionality)



## Checklist:

-   [x] All Comments addressed or cited below.
-   [x] https://github.com/0xProject/0x-monorepo/pull/2541#discussion_r405837623
-   [ ] https://github.com/0xProject/0x-monorepo/pull/2541#pullrequestreview-391267634
